README.md

```diff
@@ -1,16 +1,139 @@
 # Exercises
 
-__NOTE:__ DO NOT USE THE BUILT IN METHODS WHEN ASKED TO MAKE THEM! 
+__NOTE:__ DO NOT USE THE BUILT IN METHODS WHEN ASKED TO MAKE THEM!
 
-1. Using Array.prototype add a new method `myForEach` that mimics the behaivor of the built in `forEach`.
+//1. Using Array.prototype add a new method `myForEach` that mimics the behaivor of the built in `forEach`.
+Array.prototype.myForEach = function (callback) {
+  for(let i = 0; i < this.length; i++){
+    callback(this[i]);
+  }
 
-2. Use your `myForEach` to add `myMap` to the Array prototype. `myMap` should behave the same as regular `map`. 
+};
 
-3. Write a 'myFilter` that behave's the same as `filter`. 
+[1,2,3,4,5].myForEach((el) => {
+  console.log(el)
+});
+//2. Use your `myForEach` to add `myMap` to the Array prototype. `myMap` should behave the same as regular `map`.
+Array.prototype.myMap = function(callback) {
+let newArray = [];
+this.myForEach((el) => {
+  newArray.push(callback(el))
+})
+return newArray
+};
 
-4. Write a `myEvery`. 
 
-5. Write a `myReduce`. It should use the first element in the array if none is given. 
+let arr2 = [1,2,3,4,5]
+let finalArray = arr2.myMap((el) => {
+  return el * 2
+
+});
+
+console.log(finalArray)
+
+
+
+/*teachers response
+let arr = [1,2,3]
+function double(el){
+  return el * 2;
+}
+
+let doubled = arr.map(double)
+console.log(doubled);*/
+
+//3. Write a 'myFilter` that behave's the same as `filter`.
+/*
+Array.prototype.myFilter = function(callback){
+  let newArray = [];
+  this.myMap((el) => {
+
+    if(el % 2 === 0){
+
+    return newArray.push(callback(el))
+        }else if(el % 3 === 1){
+
+    return newArray.push(callback(el))
+  };
+
+    });
+
+  };
+  */ mine
+
+  /*working none
+  Array.prototype.myFilter = function(callback){
+  let filArr = [];
+  this.myforEach(el => {
+    if(callback){
```

Comment:
Ah! Okay. So filter doesn't actually change any element, so you wouldn't be pushing a callback'ed element to your answer array. Instead you use callback to determine *whether or not* you should push into your answer array.